hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/UtilVerbs.java

```diff
@@ -1670,20 +1674,71 @@ public static Tuple nftTransfer(
     }
 
     public static List<HapiSpecOperation> convertHapiCallsToEthereumCalls(
-            final List<HapiSpecOperation> ops, final String privateKeyRef, final Key adminKey, final long defaultGas) {
+            final List<HapiSpecOperation> ops,
+            final String privateKeyRef,
+            final Key adminKey,
+            final long defaultGas,
+            final HapiSpec spec) {
         final var convertedOps = new ArrayList<HapiSpecOperation>(ops.size());
         for (final var op : ops) {
-            if (op instanceof HapiContractCall callOp && callOp.isConvertableToEthCall()) {
+            if (op instanceof HapiContractCall callOp && callOp.isConvertableToEthCall(spec)) {
+                // if we have function params, try to swap the long zero address with the EVM address
+                if (callOp.getParams().length > 0 && callOp.getAbi() != null) {
```

Comment:
I'm not sure I understand this comment. Do you suggest creating a new class `HapiBaseContractCall` that will sit between `HapiContractCall` and `HapiBaseCall`? So `HapiContractCall` will extend `HapiBaseContractCall` and `HapiBaseContractCall` will extend `HapiBaseCall`? I am also confused about the `Optional` comment for the abi and the params. Currently, they are String and Object[] - do you suggest making them as `Optional`?

Reply:
No I completely screwed that comment up.  Let me try again.

`HapiContractCall` derives from `HapiBaseCall` which currently has both `params` and `abi` as nullable fields, rather than wrapped in an `Optional`.  

That's different from `HapiContractCreate` which derives from `HapiBaseContractCreate` where all the interesting fields _are_ wrapped in an `Optional`, including `abi` and `args`.

So I'm suggesting fixing up `HapiBaseContractCreate` to use `Optionals` too.  I think the resulting code reads better, when those fields are used using the `Optional` accessors.

Reply:
Gotcha 👍